backend/open_webui/models/users.py

```
@@ -300,5 +339,105 @@ def get_user_api_key_by_id(self, id: str) -> Optional[str]:
+                time.mktime(time.strptime(f"{today} 00:00:00", "%Y-%m-%d %H:%M:%S"))
+            )
+
+            for day in range(days):
+                day_start = today_midnight - (day * 24 * 60 * 60)
+                date_str = time.strftime("%Y-%m-%d", time.localtime(day_start))
+                fallback.append({"date": date_str, "count": 0})
+
+            return sorted(fallback, key=lambda x: x["date"])
+
+    def get_range_metrics(
+        self, start_timestamp: int, end_timestamp: int, domain: str = None
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)